### PR TITLE
Implemented functionality to convert the test result format into JSON

### PR DESCRIPTION
**Pull request by luis-gasparschroeder**
This PR enables the functionality of outputting the test results generated by Test-AzTemplate and Test-AzMarketplacePackage in JSON format.

The MCISV team has requested this feature.

You can try it out with the following command(s):
`Test-AzTemplate -TemplatePath <path> | Format-JSON`
or
`Test-AzMarketplacePackage -TemplatePath <path> | Format-JSON`

### CODE REVIEW

**Comment by luis-gasparschroeder** [comment]
@microsoft-github-policy-service agree [company="{Microsoft}"]

**Comment by luis-gasparschroeder** [comment]
This image shows to output format:
Format Option (json-format-piping)

This image shows a second output format option (implemented in this [branch](https://github.com/luis-gasparschroeder/arm-ttk/tree/json-format-piping-extended)):
Format Option (json-format-piping-extended)

Please let me know if you prefer the second option.

File: arm-ttk/Format-Json.ps1 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            $TestCase.optional = $true
+            $TestCase.message = "Unknown error in template file " + $TestResult.file.name
+        }
+
+        $TestCases += $TestCase
+    }
+
+    End {
+        # Convert the array of hashtables to JSON
+        $JSON = $TestCases | ConvertTo-Json -Depth 5
```
Comment (psah434) [inline arm-ttk/Format-Json.ps1:50]: Why depth 5?

Comment (psah434) [inline arm-ttk/Format-Json.ps1:50]: What if you just take the output and do a ConvertTO-Json?

Comment (luis-gasparschroeder) [reply arm-ttk/Format-Json.ps1:50]: See comment below

Comment (luis-gasparschroeder) [reply arm-ttk/Format-Json.ps1:50]: I thought it is required to explicitly define the Depth - fortunately it is not required, thank you for pointing that out!

**Pull request merged**